osu.Server.Queues.ScorePump/ImportHighScores.cs

```diff
@@ -97,6 +97,9 @@ public int OnExecute(CancellationToken cancellationToken)
                     {
                         soloScore.id = db.Insert(soloScore, transaction);
 
+                        // Update data to match the row ID.
+                        db.Execute("UPDATE solo_scores s SET s.data = JSON_SET(s.data, '$.id', s.id) WHERE s.id = @id", soloScore, transaction);
```

Comment:
```suggestion
                        db.Execute("UPDATE solo_scores s SET s.data = JSON_SET(s.data, '$.id', @id) WHERE s.id = @id", soloScore, transaction);
```

I'd feel more comfortable with this if it works.

Reply:
Works.